wgpu-types/src/lib.rs

```diff
@@ -2746,18 +2766,27 @@ pub struct ImageDataLayout {
     /// Offset into the buffer that is the start of the texture. Must be a multiple of texture block size.
     /// For non-compressed textures, this is 1.
     pub offset: BufferAddress,
-    /// Bytes per "row" of the image. This represents one row of pixels in the x direction. Compressed
-    /// textures include multiple rows of pixels in each "row".
-    /// Required if there are multiple rows (i.e. height or depth is more than one pixel or pixel block for compressed textures)
+    /// Bytes per "row" in an image.
     ///
-    /// Must be a multiple of 256 for [`CommandEncoder::copy_buffer_to_texture`] and [`CommandEncoder::copy_texture_to_buffer`].
+    /// A row is one row of pixels or of compressed blocks in the x direction.
+    /// This value is required if there are multiple rows (i.e. height or depth is more than one pixel or pixel block for compressed textures)
+    ///
+    /// Must be a multiple of 256 for [`CommandEncoder::copy_buffer_to_texture`] and [`CommandEncoder::copy_texture_to_buffer`]. You must manually pad
+    /// the image such that this is a multiple of 256. It will not affect the image data.
+    /// 
     /// [`Queue::write_texture`] does not have this requirement.
     ///
     /// Must be a multiple of the texture block size. For non-compressed textures, this is 1.
     pub bytes_per_row: Option<NonZeroU32>,
-    /// Rows that make up a single "image". Each "image" is one layer in the z direction of a 3D image. May be larger
-    /// than `copy_size.y`.
-    /// Required if there are multiple images (i.e. the depth is more than one)
+    /// "Rows" that make up a single "image". 
+    ///
+    /// See documentation of [`bytes_per_row`] for how a row is defined.
+    ///
+    /// An image is one layer in the z direction of a 3D image or 2DArray texture. 
+    ///
+    /// The amount of rows per image may be larger than the actual amount of rows of data.
```

Comment:
need to talk about texel blocks as well

Reply:
What do you mean by this? I defined what definition of "row" I was using a line or two above.

Reply:
Oh, you mean this?
> A row is one row of pixels or of compressed blocks in the x direction.

Since it's documenting a different field, the docs do not obviously apply here. Let's be more explicit, it's ok to be redundant for this.